scipy/spatial/distance.py

```diff
@@ -1090,10 +1109,7 @@ def chebyshev(u, v, w=None):
     v = _validate_vector(v)
     if w is not None:
         w = _validate_weights(w)
-        has_weight = w > 0
-        if has_weight.sum() < w.size:
-            u = u[has_weight]
-            v = v[has_weight]
+        return max((w > 0) * abs(u - v))
```

Comment:
So, the previous behavior seems to be that weights are only used if `w` has at least one value that is `<=0` and at least one value that is `>0`. That's already a bit weird in my mind--why wouldn't it be valid to say, use the weight at their exact floating point values if they together sum to unity but none of them are zero?

Does the previous behavior actually *weight* the values? I understand that it does seem to be using `w`, but it looks more like a "mask" that only keeps dimensions with a positive weight, and doesn't actually *weight* that dimension by its floating-point weight value.

In the new scheme, we're still effectively "masking" rather than "weighting" aren't we? Is your argument that this "masking" on `w > 0` falls out naturally from the L-infinity norm? For the intuitive "chessboard distance" interpretation, my brain doesn't seem to have a problem with the concept of simply increasing/decreasing the costs of `x` and `y` directions/movements by applying floating value weights, which I think we would do with i.e., plain L2 norm?

To be more concrete, is it really intuitive that `w` arrays of `[0.17, 0.83]`, `[0.25, 0.75]` would be treated as the "same weights?"